lightning/src/chain/onchaintx.rs

```diff
@@ -733,31 +733,14 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 						// outpoints to know if transaction is the original claim or a bumped one issued
 						// by us.
 						let mut are_sets_equal = true;
-						if !request.requires_external_funding() || !request.is_malleable() {
-							// If the claim does not require external funds to be allocated through
-							// additional inputs we can simply check the inputs in order as they
-							// cannot change under us.
-							if request.outpoints().len() != tx.input.len() {
+						// Do a linear search for each input (we don't expect large input sets to
+						// exist) to ensure the request's input set is fully spent to be resilient
+						// against the external claim reordering inputs.
```

Comment:
I think here we can assume we have counterparty malleable transactions, where only one input is claiming a HTLC output, all the remaining ones can be freely malleable (even sounds true pre-anchor, HTLC-preimage on holder commitment transaction). Therefore, you might have to assume transactions with thousands of inputs, and the worst-case reasoning would be rather the block size. If you assume something like 41 bytes for the outpoint/nsequence/script_sig and 1 WU for the witnesscript (let's say an OP_TRUE), you might have each input taking 165 WU, and for a block ~24k of inputs to iter on ? So inner iterations would be more like 576M. Even if this reasoning is more accurate, and you could really stale a LDK node on this, I think this a low-concern as your counterparty would have to buy the whole blockspace, and keep going on.

Reply:
From my understanding, if the iteration is based on the transaction matched inputs, the counterparty could influence it. Though, as mentioned I don't think this is a real concern as we're ultimately bounded by the block size here. Effectively, not so bad to be more conservative here, if we modify our aggregation logic to enable cross-channel one, we could have more than 483*2 input in a transaction issued by ourselves. We might do so in the future to save on fees.